src/Microsoft.Identity.Client/AppConfig/AzureCloud.cs

```diff
@@ -0,0 +1,49 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// </summary>
+    public enum AzureCloud
+    {
+        /// <summary>
+        ///     Maps to https://login.microsoftonline.com
+        /// </summary>
+        AzurePublic,
+
+        /// <summary>
+        ///     Maps to https://login.chinacloudapi.cn
+        /// </summary>
+        AzureChina,
+
+        /// <summary>
+        ///     Maps to https://login.microsoftonline.de
+        /// </summary>
+        AzureGermany
+    };
```

Comment:
+AzureUsGovernment
@negoe : do we need to add another one?

Reply:
there are two different US gov clouds, does this take both into account?

---
In reply to: [245865575](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/761#discussion_r245865575) [](ancestors = 245865575)

Reply:
This is pending discussion from JM, and we can postpone finalizing this from this PR.

---
In reply to: [246487776](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/761#discussion_r246487776) [](ancestors = 246487776,245865575)

Reply:
@jmprieur Yes we so. Initially the Azure Government was getting mapped to login.microsoftonline.com that is why probably it is not listed here. But now as we have separate identity stack for US Gov we have separate authority i.e. login.microsoftonline.us so we do need to update it. 

@henrik-me this will work for both, we only have one authority for both GCC-high and GCC.